### PR TITLE
[ORCA-226] Map Biospecimen IDs

### PR DESCRIPTION
**Pull request by BWMac**
This PR updates the `immune_subtype_clustering` scripts and README to use the term "classifier" instead to align with the name change currently being implemented in the `iatlas-workflows` repository ([PR](https://github.com/CRI-iAtlas/iatlas-workflows/pull/28)). 

In order to meet the requirements for running the HTAN OHSU data through this workflow, I updated the `prepare_data_sheet.py` script to retrieve the biospecimen id for each input data file, and replaced the file name id that we were initially using. The final output file now has the new ids as column headers. I tested the updated workflow on Tower and the new output has been uploaded to [Synapse](https://www.synapse.org/#!Synapse:syn51475463).

### CODE REVIEW

File: immune_subtype_classifier/prepare_data_sheet.py [hunk 1]
```diff
@@ -92,11 +106,14 @@ def syn_upload(
...
-    export_name = sys.argv[2]  # "immune_subtype_sample_sheet.tsv"
-    upload_location = sys.argv[3]  # "syn51471781"
+    map_file_id = sys.argv[2]  # "syn51526489"
+    export_name = sys.argv[3]  # "immune_subtype_sample_sheet.tsv"
+    upload_location = sys.argv[4]  # "syn51526284"
     syn = syn_login()
     file_entity_list = download_data_files(parent=parent, syn=syn)
-    gene_df_list = load_data_files(file_entity_list=file_entity_list)
+    gene_df_list = load_data_files(
+        file_entity_list=file_entity_list, map_file_id=map_file_id, syn=syn
```
Comment (thomasyu888) [inline immune_subtype_classifier/prepare_data_sheet.py:115]: Can you add the map_file_id to the provenance so we can track it?

**Review by thomasyu888 — CHANGES_REQUESTED** [review]
Nice work! Just some comments to address prior to approval.

File: immune_subtype_classifier/prepare_data_sheet.py [hunk 2]
```diff
@@ -40,13 +49,18 @@ def transform_df(df: pd.DataFrame, patient_id: str) -> pd.DataFrame:
...
+    map_file_id: str,
+    syn: synapseclient.Synapse,
+) -> List[pd.DataFrame]:
     """loads all files in data_diectory into dataframes, creates Hugo column,
     grabs Hugo and tpm columns, renames tppm column to patient id, appends each df to gene_df_list
     """
     gene_df_list = []
     for file in file_entity_list:
-        patient_id = file.name.split("_")[0]
+        file_id = file.annotations.get("HTANDataFileID")[0]
```
[the remark this answers is not in the document]
Comment (BWMac) [reply immune_subtype_classifier/prepare_data_sheet.py:62]: Yeah, only the data file id is

File: immune_subtype_classifier/prepare_data_sheet.py (hunk 1 above)
Comment (BWMac) [reply immune_subtype_classifier/prepare_data_sheet.py:118]: added in latest commit

File: immune_subtype_classifier/prepare_data_sheet.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (thomasyu888) [reply immune_subtype_classifier/prepare_data_sheet.py:62]: Can you post on Iatlas channel about this with an example link to an entity? Ideally the biospecimen id is an annotation, but no need to wait for them to be annotated.

**Review by thomasyu888 — APPROVED** [review]
🦖 LGTM! I'm going to pre-approve but please see comments.

File: immune_subtype_classifier/prepare_data_sheet.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (BWMac) [reply immune_subtype_classifier/prepare_data_sheet.py:62]: posted.

**Pull request merged**